conda_rpms/install.py

```diff
@@ -516,18 +643,42 @@ def link(pkgs_dir, prefix, dist, linktype=LINK_HARD, index=None, target_prefix=N
         return
 
     source_dir = join(pkgs_dir, dist)
-    if not run_script(dist, 'pre-link', prefix, target_prefix):
+    if not run_script(prefix, dist, 'pre-link', target_prefix):
         sys.exit('Error: pre-link failed: %s' % dist)
 
     info_dir = join(source_dir, 'info')
     files = list(yield_lines(join(info_dir, 'files')))
     has_prefix_files = read_has_prefix(join(info_dir, 'has_prefix'))
     no_link = read_no_link(info_dir)
 
+    noarch = False
+    # If the distribution is noarch, it will contain a `link.json` file in
+    # the info_dir
+    link_json = join(info_dir, 'link.json')
+    if exists(link_json):
+        with open(link_json, 'r') as fh:
+            link_data = json.loads(fh.read())
+        if 'noarch' in link_data:
+            noarch_json = link_data['noarch']
+            noarch = noarch_json['type']
+
+            target_py_version = get_python_version(prefix)
+            target_python_short_path = join('bin', 'python{}'.format(
+                target_py_version))
+            target_site_packages = join('lib', 'python{}'.format(
+                target_py_version), 'site-packages') # not supporting windows?
+
     with Locked(prefix), Locked(pkgs_dir):
         for f in files:
             src = join(source_dir, f)
-            dst = join(prefix, f)
+
+            if noarch == 'python':
+                noarch_f = get_python_noarch_target_path(f,
+                                                         target_site_packages)
+                dst = join(prefix, noarch_f)
+            else:
+                dst = join(prefix, f)
```

Comment:
Is this really all we need to do for ```noarch: whatever``` packages? Can we be a bit more defensive on this?

Reply:
Not sure what you mean by this